apps/backend/src/app/modules/session/run/run-processor.class.ts

```diff
@@ -154,159 +191,176 @@ export class RunProcessor {
     { checkpoints, checkpointsRequired, checkpointsOrdered }: Segment,
     timestamps: RunSessionTimestamp[]
   ) {
-    // First checkpoint is always the start zone. It's never possible to skip a
+    // First minor is always the start zone. It's never possible to skip a
     // start zone.
-    if (timestamps[0].checkpoint !== 0) {
+    if (timestamps[0].minorNum !== 1) {
       throw new RunValidationError(ErrorType.BAD_TIMESTAMPS);
     }
 
     if (checkpointsOrdered) {
       for (let i = 1; i < timestamps.length; i++) {
-        if (timestamps[i].checkpoint <= timestamps[i - 1].checkpoint) {
+        if (timestamps[i].minorNum <= timestamps[i - 1].minorNum) {
```

Comment:
I think there also needs to be a check for all unique `minorNum` when `checkpointsOrdered === false`

Reply:
Duplicate `minorNum`s should *never* be possible, so I check for them `validateSessionTimestamps` already. Added a test for the case you give to make sure, which already passes.